examples/graph/mpnn-molecular-graphs.py

```diff
@@ -0,0 +1,723 @@
+traditional approaches often operate on precomputed molecular features such as
+molecular weight, polarity, charge, number of carbon atoms, etc. Although these
+molecular features prove to be good predictors for various molecular properties, it is
+hypothesized that operating on these more "raw", "low-level", features could prove even
+better.*
+
+### References
+
+In recent years, a lot of effort has been put into developing graph neural networks for
+graph data, including molecular graphs. For a summary of graph neural networks, see e.g.,
+[A Comprehensive Survey on Graph Neural Networks](https://arxiv.org/abs/1901.00596) and
+[Graph Neural Networks: A Review of Methods and Applications](https://arxiv.org/abs/1812.08434);
+and for further reading on the specific
+graph neural network implemented in this tutorial see
+[Neural Message Passing for Quantum Chemistry](https://arxiv.org/abs/1704.01212) and
+[DeepChem's MPNNModel](https://deepchem.readthedocs.io/en/latest/api_reference/models.html#mpnnmodel).
+"""
+
+"""
+## Setup
+
+### Install RDKit and other dependencies
+
+(Text below extracted from
+[this tutorial](https://keras.io/examples/generative/wgan-graphs/))
+
+[RDKit](https://www.rdkit.org/) is a collection of cheminformatics and machine-learning
+software written in C++ and Python. In this tutorial, RDKit is used to conviently and
+efficiently transform
+[SMILES](https://en.wikipedia.org/wiki/Simplified_molecular-input_line-entry_system) to
+molecule objects, and then from those obtain sets of atoms and bonds.
+
+SMILES expresses the structure of a given molecule in the form of an ASCII string.
+The SMILES string is a compact encoding which, for smaller molecules, is relatively
+human-readable. Encoding molecules as a string both alleviates and facilitates database
+and/or web searching of a given molecule. RDKit uses algorithms to
+accurately transform a given SMILES to a molecule object, which can then
+be used to compute a great number of molecular properties/features.
+
+Notice, RDKit is commonly installed via [Conda](https://www.rdkit.org/docs/Install.html).
+However, thanks to
+[rdkit_platform_wheels](https://github.com/kuelumbus/rdkit_platform_wheels), rdkit
+can now (for the sake of this tutorial) be installed easily via pip, as follows:
+
+```
+pip -q install rdkit-pypi
+```
+
+And for easy and efficient reading of csv files and visualization, the below needs to be
+installed:
+
+```
+pip -q install pandas
+pip -q install Pillow
+pip -q install matplotlib
+pip -q install pydot
+sudo apt-get -qq install graphviz
+```
+"""
+
+"""
+### Import packages
+"""
+
+import tensorflow as tf
+from tensorflow import keras
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import warnings
+from rdkit import Chem, RDLogger
+from rdkit.Chem.Draw import IPythonConsole, MolsToGridImage
+import logging
+
+tf.get_logger().setLevel(logging.ERROR)
+warnings.filterwarnings("ignore")
```

Comment:
Are these 2 calls really necessary?

Reply:
If commenting out `tf.get_logger().setLevel(logging.ERROR)`, I get the following warning when building or compiling the model:
```
WARNING:tensorflow:From /home/alex/Envs/keras_io/lib/python3.8/site-packages/tensorflow/python/ops/array_ops.py:5043: calling gather (from tensorflow.python.ops.array_ops) with validate_indices is deprecated and will be removed in a future version.
Instructions for updating:
The `validate_indices` argument has no effect. Indices are always validated on CPU and never validated on GPU.
```

And if commenting out `warnings.filterwarnings("ignore")`, I get the following type of warning(s) when fitting the model:
```
warnings.warn(
/home/alex/Envs/keras_io/lib/python3.8/site-packages/tensorflow/python/framework/indexed_slices.py:447: UserWarning: Converting sparse IndexedSlices(IndexedSlices(indices=Tensor("gradient_tape/model/message_passing/edge_network/Reshape_4:0", shape=(None,), dtype=int32), values=Tensor("gradient_tape/model/message_passing/edge_network/Reshape_3:0", shape=(None, 64), dtype=float32), dense_shape=Tensor("gradient_tape/model/message_passing/edge_network/Cast:0", shape=(2,), dtype=int32))) to a dense Tensor of unknown shape. This may consume a large amount of memory.
```